tests/pauli_rotations/rotation_test.py

```diff
@@ -91,6 +91,37 @@ def test_pauli_rotation_eq(self, pauli_rotation_1, pauli_rotation_2):
     def test_pauli_rotation_ne(self, pauli_rotation_1, pauli_rotation_2):
         assert pauli_rotation_1 != pauli_rotation_2
 
+    @pytest.mark.parametrize(
+        "input, expected",
+        [
+            (PauliRotation.from_list([Y], Fraction(1, 2)), "1/2: [Y]"),
+            (PauliRotation.from_list([I, X], Fraction(4, 2)), "2: [I, X]"),
+            (PauliRotation.from_list([Y, X], Fraction(-0, 5)), "0: [Y, X]"),
+            (PauliRotation.from_list([X, Z, Y], Fraction(-6, 8)), "-3/4: [X, Z, Y]"),
+        ],
+    )
+    def test_str(self, input, expected):
+        assert str(input) == expected
+
+    @pytest.mark.parametrize(
+        "input, expected",
+        [
+            (PauliRotation.from_list([Y], Fraction(1, 2)), r"(Y)_\frac{\pi}{2}"),
+            (PauliRotation.from_list([Z, I], Fraction(-0, 4)), r"(Z \otimes I)_0"),
+            (PauliRotation.from_list([X, Y], Fraction(-6, 3)), r"(X \otimes Y)_-2\pi"),
```

Comment:
On mobile, can't run the tests, but shouldn't shouldn't the index script be `_{-2*pi}` looks like we might have a bug

Reply:
Since the ci passes

Reply:
I just checked with overleaf. It should be `(X \otimes Y)_{-2\pi}` so we are having a bug here.

Reply:
Just pushed a fix

Reply:
I think its good now